### PR TITLE
PHOENIX-7130 Support skipping of shade sources jar creation

### PR DESCRIPTION
**Pull request by NihalJain**

### CODE REVIEW

**Review by stoty — APPROVED** [review]
+1 LGTM

**Comment by NihalJain** [comment]
Hi @stoty are we good here? Also please let me know if should take this to 5.1. Will need to raise another PR  as seems an additional file change is needed in branch-5.1.

File: pom.xml [hunk 1]
```diff
@@ -1887,6 +1888,17 @@
           @{jacocoArgLine}</argLine>
       </properties>
     </profile>
+    <profile>
```
Comment (stoty) [inline pom.xml:1891]: Do we need a profile for this ?
Specifying -DshadeSources=false would have the same effect.

But if you  prefer this way, it's fine, too.

**Review by stoty — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (NihalJain) [reply pom.xml:1891]: Yes actually just having a profile for a variable which can be passed from outside does not make sense. But I was thinking if we also add `maven.source.skip` to be true `here` and rename this profile to `skipSourceProcessing` or something.

Comment (NihalJain) [reply pom.xml]: Hi @stoty Removed the profile as suggested by you.

**Comment by NihalJain** [comment]
Please donot push this to 5.1, have raised another PR for 5.1 as code is a bit different.

**Comment by stoty** [comment]
I have tested this now, but the shading time difference is seems to be minimal.
On my machine it takes ~3:20 with the shaded sources, and ~3:05 without them.

I don't have a problem with committing this, but I wander what kind of improvement do you see with the patch, @NihalJain  ?

Also I have just merged phoenix-client-lite, you may want to update the patch to add same change there.

**Comment by NihalJain** [comment]
> I have tested this now, but the shading time difference is seems to be minimal. On my machine it takes ~3:20 with the shaded sources, and ~3:05 without them.

In our internal jenkins, which is based on branch-5.1, the difference is substantial. Maybe because we download all jars for every build and also possibly our local artifactory + VPN is not as fast, so downloading the source jars take almost 33% of build time.

Without  `-PskipShadeSources -Dmaven.source.skip=true`
```
.
.
[INFO] Phoenix Client ..................................... SUCCESS [22:40 min]
[INFO] Phoenix Client Embedded ............................ SUCCESS [08:56 min]
[INFO] Phoenix Server JAR ................................. SUCCESS [01:54 min]
[INFO] Phoenix Assembly ................................... SUCCESS [ 38.548 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  36:06 min
[INFO] Finished at: 2024-01-03T10:55:52Z
[INFO] ------------------------------------------------------------------------
```

Vs 

With  `-PskipShadeSources -Dmaven.source.skip=true`
```
.
.

[INFO] Phoenix Client ..................................... SUCCESS [09:37 min]
[INFO] Phoenix Client Embedded ............................ SUCCESS [08:14 min]
[INFO] Phoenix Server JAR ................................. SUCCESS [01:49 min]
[INFO] Phoenix Assembly ................................... SUCCESS [ 38.593 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  24:08 min
[INFO] Finished at: 2024-01-03T11:29:24Z
[INFO] ------------------------------------------------------------------------
```


> Also I have just merged phoenix-client-lite, you may want to update the patch to add same change there.

Sure let me rebase and make appropriate changes, if any

**Comment by stoty** [comment]
Ouch.
That's extremley slow.
Is it on a spinning rust HDD ? Or some slow NAS ?

**Review by gjacoby126 — APPROVED** [review]
+1

**Comment by gjacoby126** [comment]
(Note that we're waiting on #1736 before merging since it's a blocker.)

**Comment by stoty** [comment]
This is a pom only change, and #1736 doesn't touch the poms, so I wouldn't expect them to confict in any way.

**Comment by virajjasani** [comment]
I think it's fine to merge this since it adds the option to skip the shaded jar creation.

**Review by virajjasani — APPROVED** [review]

**Comment by stoty** [comment]
Also, I'm going to merge the phoenix-server shading refactor soon (another pom only change), It would probably be a good idea to  add this change there as well.

**Comment by virajjasani** [comment]
pom changes are fine so far, but yeah any other big changes or feature changes that have potential to cause any additional issues (more than what is known with #1736 and make it worse) are potential blockers e.g. JSON support, which is technically ready for merge (though still in final review phase) but i would like to block it.

**Comment by NihalJain** [comment]
Rebased code and updated all new instances of createSourcesJar in code. Could you please review and merge this.?

CC: @gjacoby126 @stoty @virajjasani

**Review by virajjasani — APPROVED** [review]
+1

**Comment by chrajeshbabu** [comment]
Since build failures are not related this. Merging

**Pull request merged**